File: QsWebSoft/Xt_Popwin/W_Hssh_Tsl.win.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using System.Web;
using System.Data;
using System.Data.SqlClient;
using Sybase.DataWindow;
using TXSoft.Common;
using TXSoft.ExtPB;
using TXSoft.DataStore;
using System.Linq;
using System.IO;
using Microsoft.Win32;
using System.Configuration;
using System.Diagnostics;
using QsWebSoft.Helper;


namespace QsWebSoft.Xt_Popwin
{
    public partial class W_Hssh_Tsl
    {
        public W_Hssh_Tsl()
        {

        }
        public override void OnLoad()
        {
            base.OnLoad();
            ReportService report = (ReportService)dw_1.Services.Add(ServiceName.Report);
            report.RequestorDrawTitle = false;
            var method = "getTaxRate";
            var code = "5407520091";
            //var code = this.Request["code"].ToString(); 
            //var tsflag = this.Request["tsflag"].ToString(); 

            string strFile = AppDomain.CurrentDomain.BaseDirectory;
            var timestamp = Context.Timestamp.ToString("yyyy-MM-dd HH:mm:ss");
            string sign = WS.MD5("hd01" + "750a8953a27a4d0aa6b4a70576574ead" + timestamp);
            //method = "getVisualLogistics";
            string format = "xml";
            string uid = "hd01";
            string url = "";
            //取随机数
            Random rnd = new Random();
            int n = rnd.Next(10000, 99999);
            url = "http://api.trainer.nbeport.com/router/rest?method=" + method + "&format=" + format + "&code=" + code + "&user_id=" + uid + "&timestamp=" + timestamp + "&sign=" + sign + "&t=" + n;
            //url = Server.UrlEncode(url);
            //WriteTextLog(url);
            strFile = strFile + "XML\\" +  code + ".xml";

            //WS.DownLoadAndCreateXML(url, strFile);
            var Stream = WS.WriteCardToStream(url);
            //STREAM转换为string
            StreamReader reader = new StreamReader(Stream);
            string text = reader.ReadToEnd();
            //ds_1.ImportString(text, Sybase.DataWindow.FileSaveAsType.Xml);
            //var row = 0;
            //var sxrq = "";
            //var jsrq = "";
            //var insertrow = 0;
            //var name = "";
            //var unit = "";
            //var dwcode = "";
            //var zssl_Set = "";
            //var tsl = "";
            //if (ds_1.RowCount > 1)
            //{
            //    //更新行号
            //    for (row = ds_1.RowCount; row > 0; row--)
            //    {
            //        insertrow = dw_1.InsertRow(0);
            //        sxrq = ds_1.GetItemString(row, "vesselname");
            //        jsrq = ds_1.GetItemString(row, "voyage");
            //        if (sxrq != null && sxrq != "")
            //        {
            //            name = ds_1.GetItemString(row + 1, "stepname");
            //            unit = ds_1.GetItemString(row + 1, "remark");
            //            dwcode = ds_1.GetItemString(row + 1, "ctnno");
            //            zssl_Set = ds_1.GetItemString(row + 1, "billno");
            //            tsl = ds_1.GetItemString(row + 1, "procresul");
            //        }

            //        dw_1.SetItemString(insertrow, "vesselname", sxrq);
            //        dw_1.SetItemString(insertrow, "voyage", jsrq);
            //        dw_1.SetItemString(insertrow, "stepname", name);
            //        dw_1.SetItemString(insertrow, "remark", unit);
            //        dw_1.SetItemString(insertrow, "ctnno", dwcode);
            //        dw_1.SetItemString(insertrow, "procresul", tsl);
            //        dw_1.SetItemString(insertrow, "billno", zssl_Set);
            //    }

            //}
            //dw_1.ImportString(text, Sybase.DataWindow.FileSaveAsType.Xml);
            //dw_1.Retrieve();
            dw_1.Modify("DataWindow.Readonly=yes");
            AjaxPro.Utility.RegisterTypeForAjax(typeof(PubMethod));
            this.RegisterClientScriptInclude("W_Hssh_Tsl_Import", "/Xt_Popwin/W_Hssh_Tsl_Import.win.js");
        }
        /// <summary>   
        /// 写入日志到文本文件   
        /// </summary>   
        /// <param name="action">动作</param>   
        /// <param name="strMessage">日志内容</param>   
        /// <param name="time">时间</param>   
        public static void WriteTextLog(string timestamp)
        {
            string path = AppDomain.CurrentDomain.BaseDirectory;
            if (!Directory.Exists(path))
                Directory.CreateDirectory(path);

            string fileFullPath = path + "\\XML\\Tsl.txt";
            StringBuilder str = new StringBuilder();
            str.Append(timestamp);

            StreamWriter sw;
            if (!File.Exists(fileFullPath))
            {
                sw = File.CreateText(fileFullPath);
            }
            else
            {
                sw = File.AppendText(fileFullPath);
            }
            sw.WriteLine(str.ToString());
            sw.Close();
        } 

    }
}

File: QsWebSoft/Xt_Popwin/W_Hssh_Tsl.win.js
///<reference path="/Scripts/ExtPB_Net.js" />
///<reference path="/Scripts/MyExtPB.js" />
///<reference path="/Beta3/ExtPB_Demo.js" />

function W_Hssh_Tsl() {

	 var self=this;
	 var requestor= new PBWindow();
	 var dw_1= new DataWindow( requestor );
	 var tb_1= new ToolStrip( requestor );
	 var sle_1= new SingleLineEdit( requestor );
	 var ds_1= new DataStore( requestor );

	 if(typeof(this.bind)=="function") var baseBind=this.bind;
	 this.bind=function(win){
	 if(typeof( baseBind )=="function") baseBind(win);

		 if(requestor == null ) requestor= new PBWindow();
		 requestor.Attach(win);
		 if(dw_1 == null ) dw_1= new DataWindow( requestor );
		 dw_1.Attach( win.Control( "dw_1") ); 
		 if(tb_1 == null ) tb_1= new ToolStrip( requestor );
		 tb_1.Attach( win.Control( "tb_1") ); 
		 if(sle_1 == null ) sle_1= new SingleLineEdit( requestor );
		 sle_1.Attach( win.Control( "sle_1") ); 
		 if(ds_1 == null ) ds_1= new DataStore( requestor );
		 ds_1.Attach( win.Control( "ds_1") ); 

		 if(typeof(self.PostOpen)=="function") 
			 win.AttachEvent("window","PostOpen",self.PostOpen); 

		 if(typeof(self.OK)=="function") 
			 win.AttachEvent("dw_1","DoubleClicked",self.OK); 

		 if(typeof(self.OK)=="function") 
			 win.AttachEvent("btn_ok","Clicked",self.OK); 

		 if(typeof(self.Exit)=="function") 
			 win.AttachEvent("btn_Cancel","Clicked",self.Exit); 

	 }

	 if(typeof(this.unbind)=="function") var baseUnBind=this.unbind; 
	 this.unbind=function() {
	 if(typeof(this.Release)=="function") this.Release(); 
	 if(typeof(baseUnBind)=="function") baseUnBind(); 

		 dw_1.Detach(); dw_1=null; 
		 tb_1.Detach(); tb_1=null; 
		 sle_1.Detach(); sle_1=null; 
		 ds_1.Detach(); ds_1=null; 
		 requestor.Detach(); requestor=null; 

		 setTimeout(CollectGarbage, 100); 
	 }

	 //以上代码由设计器自动生成,请不要手工修改
	 //ExtPB.Net JavaScript End  设计器查找标志,请不要删除或修改

    var dw_mx = null;

    this.SetDataWindow = function (dw) {
        dw_mx = dw;
    }
    var column = null;
    this.SetData = function (parm) {
        column = parm;
    }


    //#region 窗口的OPEN事件
    this.PostOpen = function () {
        sle_1.SetFocus();
    }
    this.OK = function () {

        if (sle_1.GetText() == "" || sle_1.GetText() == null) {
            alert("请输入要查询的海关编码!");
            return;
        }
        //        var format = "xml";
        //        var uid = "hd01";
        //        var method = "getTaxRate";
        //        var code = "5407520091";
        //        var url = "http://api.trainer.nbeport.com/router/rest?method=" + method + "&format=" + format + "&code=" + code + "&user_id=" + uid;
        //        var xml = QsWebSoft.PubMethod.GetXml(url, code).value;
        ////        ds_1.ImportString(xml);

        ////                dw_1.ImportString(xml);
        //                dw_1.ImportStringEx(15,xml);
        dw_1.Reset();
        var code = sle_1.GetText();
        var iw_Hssh_Import = null;
        if (iw_Hssh_Import == null)
            iw_Hssh_Import = new W_Hssh_Tsl_Import();
        iw_Hssh_Import.SetDataWindow(dw_1);

        requestor.Open("/LoadWindow.ashx?WinName=Xt_Popwin.W_Hssh_Tsl_Import&code=" + code, iw_Hssh_Import);



    }

    this.Exit = function () {
        setTimeout("window.close();", 100);
    }

   


 }